pkg/globalnet/controllers/global_ingressip_controller.go

```diff
@@ -125,9 +134,29 @@ func (c *globalIngressIPController) onCreate(ingressIP *submarinerv1.GlobalIngre
 		return true
 	}
 
-	// TODO - remove IP table rules for previous allocated IP
+	if ingressIP.Spec.Target == submarinerv1.ClusterIPService && ingressIP.Spec.ServiceRef != nil {
```

Comment:
`ingressIP.Spec.ServiceRef` isn't used in this block - is there a specific reason to check it?

```suggestion
	if ingressIP.Spec.Target == submarinerv1.ClusterIPService {
```